airflow/executors/kubernetes_executor.py

```diff
@@ -599,6 +608,9 @@ def sync(self) -> None:
         for _ in range(self.kube_config.worker_pods_creation_batch_size):
             try:
                 task = self.task_queue.get_nowait()
+                if datetime.utcnow() < task.next_allowed_retry:
+                    continue
```

Comment:
Not sure how this will work together with the self.event_scheduler.run(blocking=False) on line 646 below. Will it keep looping very fast as long as the queue has items or is there some other delay built in here?